### PR TITLE
Cucumber tests mealplan

### PR DESCRIPTION
**Pull request by ahz0001**
Added cucumber tests and step definitions.  Put meal plan code block into its own function in UsersController.

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -131,6 +116,25 @@ def show
...
+		days = {1 => "Monday", 2 => "Tuesday", 3 => "Wednesday",
+			4 => "Thursday", 5 => "Friday", 6 => "Saturday", 7 => "Sunday"}
+
+		all_recipes['items'].each do |recipe|
+			if !daily_recipes.key?(recipe['slot'])
+				daily_recipes[recipe['slot']] = Hash.new
+			end
+			if !daily_recipes[recipe['slot']].key?(days[recipe['day']])
+				daily_recipes[recipe['slot']][days[recipe['day']]] = Hash.new
+				daily_recipes[recipe['slot']][days[recipe['day']]] = recipe
```
Comment (cheolhojang) [inline app/controllers/users_controller.rb:132]: good catch on the indentation

**Review by cheolhojang — COMMENTED** [review]
Good changes, solid work

File: features/filter_recipes.feature [hunk 2]
```diff
@@ -7,52 +7,46 @@ Feature: Display filtered recipes to user sorted into Breakfast, Lunch & Dinner
...
+  Then I should see "Mushroom Risotto" in dinner
+    And I should see "Burrito" in dinner
+    And I should see "Milk" in dinner
 
 Scenario: API call returns no breakfast recipes
-  When I view the meals page
-    And view the Breakfast section
-  Then I should see an empty section
+  When API call returns no breakfast recipes
+  Then I should see nothing in breakfast
```
Comment (nkompella) [inline features/filter_recipes.feature:44]: In this Scenario and the other two (returns no lunch recipes, returns no dinner recipes), maybe include the steps for "should see lunch" and "should see dinner"

Comment (ahz0001) [reply features/filter_recipes.feature:44]: resolved

File: app/controllers/users_controller.rb [hunk 3]
```diff
@@ -131,6 +116,25 @@ def show
 		@dinPrice = (@dinHash["price"] / 100).round(2)
 	end
 
+	def self.do_daily_recipes(all_recipes)
```
Comment (nkompella) [inline app/controllers/users_controller.rb:119]: I like that you modularized this by moving code to a separate function!

**Pull request merged**